storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6356,31 +6356,27 @@ void ha_rocksdb::free_key_buffers() {
 }
 
 void ha_rocksdb::set_use_read_free_rpl(const char *const whitelist) {
-  DBUG_ASSERT(whitelist != nullptr);
-
 #if defined(HAVE_PSI_INTERFACE)
   Regex_list_handler regex_handler(key_rwlock_read_free_rpl_tables);
 #else
   Regex_list_handler regex_handler;
 #endif
 
-  if (!regex_handler.set_patterns(whitelist)) {
+  if (!regex_handler.set_patterns(whitelist ? whitelist : "")) {
     warn_about_bad_patterns(&regex_handler, "read_free_rpl_tables");
   }
 
   m_use_read_free_rpl = regex_handler.matches(m_tbl_def->base_tablename());
 }
 
 void ha_rocksdb::set_skip_unique_check_tables(const char *const whitelist) {
-  DBUG_ASSERT(whitelist != nullptr);
-
 #if defined(HAVE_PSI_INTERFACE)
   Regex_list_handler regex_handler(key_rwlock_skip_unique_check_tables);
 #else
   Regex_list_handler regex_handler;
 #endif
 
-  if (!regex_handler.set_patterns(whitelist)) {
+  if (!regex_handler.set_patterns(whitelist ? whitelist : ".*")) {
```

Comment:
Is this expected behaviour? I would think that setting to null would mean that no tables are going to skip unique checks?

Reply:
This is a bit confusing, since ".*" is defined as the default value for `skip_unique_check_tables` (https://github.com/facebook/mysql-5.6/blob/fb-mysql-5.6.35/storage/rocksdb/ha_rocksdb.cc#L757). Setting a parameter's value to NULL, where NULL is not an accepted value, should fall back to the default value.